### ISSUE TITLE
[API] Support of loss.backward() e2e - running model backward graph

### ISSUE THREAD

**Issue by nvukobratTT** [tenstorrent/tt-forge-fe#177]
## Summary
Following torch philosophy, and making out API easier for other ML engineers to understand, next effort should be to explore if we can map `loss.backward()` to our runtime. 

## Details
All details are already explained on this issue tenstorrent/tt-forge-fe#176. The main point here is to explore how can we map loss objects to invoke a backward pass of the model.
